lib/internal/fs/glob.js

```diff
@@ -331,10 +331,7 @@ class Glob {
     const fullpath = resolve(this.#root, path);
 
     // If path is a directory, add trailing slash and test patterns again.
-    // TODO(Trott): Would running #isExcluded() first and checking isDirectory() only
-    // if it matches be more performant in the typical use case? #isExcluded()
-    // is often ()=>false which is about as optimizable as a function gets.
-    if (this.#cache.statSync(fullpath).isDirectory() && this.#isExcluded(`${fullpath}/`)) {
+    if (this.#isExcluded(`${fullpath}/`) && this.#cache.statSync(fullpath).isDirectory()) {
```

Comment:
I think the fastest way to would be to use this function: https://github.com/nodejs/node/blob/main/src/node_file.cc#L1053

Particularly: internalBinding('fs').internalModuleStat(path) which avoids creating unnecessary Dir/Dirent objects.